src/test/java/org/broadinstitute/consent/http/service/DarCollectionServiceTest.java

```diff
@@ -621,6 +633,151 @@ public void testQueryCollectionsByFiltersAndUserRolesForResearcher() {
   }
 
+  @Test
+  public void testDeleteAsResearcherNoElections() {
+    User user = new User();
+    user.setUserId(1);
+    String dacRoleName = UserRoles.RESEARCHER.getRoleName();
+    Integer dacRoleId = UserRoles.RESEARCHER.getRoleId();
+    UserRole memberRole = new UserRole(dacRoleId, dacRoleName);
+    user.addRole(memberRole);
+
+    Set<Dataset> datasets = new HashSet<>();
+    DarCollection collection = generateMockDarCollection(datasets);
+    collection.setDarCollectionId(10);
+    collection.setCreateUserId(user.getUserId());
+
+    when(electionDAO.findElectionsByReferenceIds(any())).thenReturn(new ArrayList<>());
+    when(darCollectionDAO.findDARCollectionByCollectionId(any())).thenReturn(collection);
+    List<String> referenceIds = collection.getDars().values().stream().map(DataAccessRequest::getReferenceId).collect(Collectors.toList());
+
+    Integer collectionId = collection.getDarCollectionId();
+
+    spy(electionDAO);
+    spy(dataAccessRequestDAO);
+    spy(darCollectionDAO);
+
+    initService();
+    System.out.println(collectionId);
+    service.deleteByCollectionId(user, collectionId);
+
+    // verify each DAR was deleted
+    verify(dataAccessRequestDAO, times(1)).deleteByReferenceIds(any());
+    verify(dataAccessRequestDAO, times(1)).deleteDARDatasetRelationByReferenceIds(any());
+    verify(matchDAO, times(1)).deleteMatchesByPurposeIds(any());
+    // verify overarching collection was deleted
+    verify(darCollectionDAO, times(1)).deleteByCollectionId(collectionId);
+    verify(electionDAO, times(0)).deleteElectionsFromAccessRPs(any());
+    verify(electionDAO, times(0)).deleteElectionsByIds(any());
+    verify(voteDAO, times(0)).deleteVotesByReferenceIds(any());
+
+  }
+
+  @Test
+  public void testDeleteAsResearcherWithElections() {
+    User user = new User();
+    user.setUserId(1);
+    String dacRoleName = UserRoles.RESEARCHER.getRoleName();
+    Integer dacRoleId = UserRoles.RESEARCHER.getRoleId();
+    UserRole memberRole = new UserRole(dacRoleId, dacRoleName);
+    user.addRole(memberRole);
+
+    Set<Dataset> datasets = new HashSet<>();
+    DarCollection collection = generateMockDarCollection(datasets);
+    collection.setDarCollectionId(10);
+    collection.setCreateUserId(user.getUserId());
+
+    Election e = createMockElection();
+    when(electionDAO.findElectionsByReferenceIds(any())).thenReturn(new ArrayList<>(){{add(e);}});
+    when(darCollectionDAO.findDARCollectionByCollectionId(any())).thenReturn(collection);
+    List<String> referenceIds = collection.getDars().values().stream().map(DataAccessRequest::getReferenceId).collect(Collectors.toList());
+
+    Integer collectionId = collection.getDarCollectionId();
+
+    // should not work since elections exist
+    thrown.expect(NotAcceptableException.class);
+    thrown.expectMessage("Cannot delete DAR with elections.");
+
+    initService();
+    service.deleteByCollectionId(user, collectionId);}
```

Comment:
Another way to do this is to add an expected exception to the `@Test` annotation. We generally don't test string content in messages - that can change and doesn't really add value to the exception class.

Reply:
ahh gotcha, changed all of them to `@Test` annotations!